### PR TITLE
Quick wins for Ocean

### PR DESCRIPTION
**Pull request by erikzaadi**
- [x] Add single integration version bump
- [x] Linked dev ocean-core
- [x] Precommit that also autoruns lint when we have changes in python files
- [x] Fix vscode debug entries
- [x] Add cookie cutter to make new integrations up to standards
- [x] Add simple unit tests 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: integrations/aws/changelog/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
+*
```
Comment (Tankilevitch) [inline integrations/aws/changelog/.gitignore:1]: add it to the cookiecutter

File: integrations/aws/pyproject.toml [hunk 2]
```diff
@@ -14,6 +14,7 @@ types-aioboto3 = "^12.4.0"
 types-aiobotocore = {extras = ["sts"], version = "^2.12.3"}
 
 [tool.poetry.group.dev.dependencies]
+port_ocean = { path = '../../', develop = true, extras = ['all'] }
```
Comment (Tankilevitch) [inline integrations/aws/pyproject.toml:17]: same?

File: integrations/aws/changelog/.gitignore (hunk 1 above)
Comment (erikzaadi) [reply integrations/aws/changelog/.gitignore:1]: Done

File: integrations/aws/pyproject.toml (hunk 2 above)
Comment (erikzaadi) [reply integrations/aws/pyproject.toml:17]: Done

File: .vscode/launch.json
Comment (Tankilevitch) [inline .vscode/launch.json:1]: is there a way to make it more generic? so we won't need to add that for each new integration

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.env.example [hunk 3]
```diff
@@ -0,0 +1,5 @@
+OCEAN__PORT__CLIENT_ID="<port-client-id>"
+OCEAN__PORT__CLIENT_SECRET="<port-client-secret>"
+OCEAN__EVENT_LISTENER={"type": "POLLING"}
+OCEAN__INTEGRATION__IDENTIFIER="my-{{ }}-integration"
+OCEAN__INTEGRATION__TYPE="aws"
```
Comment (matan84) [inline port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.env.example:5]: Maybe change from aws?

**Review by matan84 — CHANGES_REQUESTED** [review]
1 small change

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.env.example (hunk 3 above)
Comment (erikzaadi) [reply port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.env.example:5]: That's the thing though, we need it to be one of the default ones for it to actually work on the initial try :(, 
the `{{ }}` part is a bug though!

File: integrations/aws/poetry.lock
Comment (Tankilevitch) [inline integrations/aws/poetry.lock:1]: is this by mistake?

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-icon: Cookiecutter # Should be one of the available icons in Port
+icon: Blueprint # Should be one of the available icons in Port
 features:
   - type: exporter
     section: Git Providers # Should be one of the available sections in Port
     resources:
-      - kind: <ResourceName1>
-      - kind: <ResourceName2>
+      - kind: {{ cookiecutter.integration_slug }}-example-kind
+      # - kind: <ResourceName2>
```
Comment (Tankilevitch) [inline port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml:8]: remove?

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
 configurations:
-  - name: myGitToken
-    required: true
+  - name: my{{ cookiecutter.integration_slug}}Token
+    # required: true
```
Comment (Tankilevitch) [inline port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml:11]: same?

**Review by Tankilevitch — APPROVED** [review]
super nice 🚀

File: port_ocean/cli/commands/new.py [hunk 6]
```diff
@@ -37,6 +56,8 @@ def new(path: str, is_private_integration: bool) -> None:
     )
     name = result.split("/")[-1]
 
+    add_vscode_configuration(result, name)
```
Comment (yairsimantov20) [inline port_ocean/cli/commands/new.py:59]: Lets talk about this

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.env.example (hunk 3 above)
Comment (yairsimantov20) [reply port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.env.example:6]: I dont agree with this one
Lets find a better solution

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml [hunk 7]
```diff
@@ -1,14 +1,14 @@
 description: {{cookiecutter.integration_name}} integration for Port Ocean
-icon: Cookiecutter # Should be one of the available icons in Port
+icon: Blueprint # Should be one of the available icons in Port
 features:
   - type: exporter
     section: Git Providers # Should be one of the available sections in Port
```
Comment (yairsimantov20) [inline port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml:5]: Is it a git provider?

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml [hunk 8]
```diff
@@ -1,14 +1,14 @@
 description: {{cookiecutter.integration_name}} integration for Port Ocean
-icon: Cookiecutter # Should be one of the available icons in Port
```
Comment (yairsimantov20) [inline port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml:2]: Why?

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/pyproject.toml [hunk 9]
```diff
@@ -19,7 +21,7 @@ towncrier = "^23.6.0"
 [tool.towncrier]
 directory = "changelog"
 filename = "CHANGELOG.md"
-package = "port_ocean"
+package = "port_ocean.{{cookiecutter.integration_slug}}"
```
Comment (yairsimantov20) [inline port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/pyproject.toml:24]: If u r already here talk to me there is something i want to fix

**Review by yairsimantov20 — CHANGES_REQUESTED** [review]

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml (hunk 4 above)
Comment (erikzaadi) [reply port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml:8]: I wanted to add that as an example of what could be added

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml (hunk 5 above)
Comment (erikzaadi) [reply port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml:11]: ditto

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/pyproject.toml (hunk 9 above)
Comment (yairsimantov20) [reply port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/pyproject.toml:26]: just remove me

File: port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml (hunk 8 above)
Comment (yairsimantov20) [reply port_ocean/cli/cookiecutter/{{cookiecutter.integration_slug}}/.port/spec.yaml:2]: revert

**Comment by erikzaadi** [comment]
Separated to smaller PRs and merged

**Pull request closed without merging**